### PR TITLE
feat(dateFormat): Added support for ui.config closes #349

### LINKED ISSUE #349: Add uiDateFormat to ui.config
Should let people do something like:

```
module().value('ui.config', {
  date: {
    dateFormat: 'yy/mm/dd' // for ui-date which formats the input box (view)
  },
  dateFormat: 'yy/mm/dd' // for ui-date-format which formats the model
});
```

### PR DESCRIPTION
**Pull request by ProLoser**
Still lacking unit test for new features. Not sure how to specify `ui.config` for only 1 test.

### CODE REVIEW

File: modules/directives/date/date.js [hunk 1]
```diff
@@ -75,33 +75,34 @@ angular.module('ui.directives')
...
-.directive('uiDateFormat', [function() {
+.directive('uiDateFormat', ['ui.config', function(uiConfig) {
   var directive = {
     require:'ngModel',
     link: function(scope, element, attrs, modelCtrl) {
-      if ( attrs.uiDateFormat === '' ) {
-        // Default to ISO formatting
+      var dateFormat = attrs.uiDateFormat || uiConfig.dateFormat;
+      if ( dateFormat ) {
+        var format = attrs.uiDateFormat;
```
Comment (petebacondarwin) [inline modules/directives/date/date.js]: Doesn't this line defeat the object of looking up dateFormat again uiConfig?

File: modules/directives/date/date.js [hunk 2]
```diff
@@ -75,33 +75,34 @@ angular.module('ui.directives')
...
+        // Use the datepicker with the attribute value as the format string to convert to and from a string
         modelCtrl.$formatters.push(function(value) {
           if (angular.isString(value) ) {
-            return new Date(value);
+            return $.datepicker.parseDate(format, value);
           }
         });
         modelCtrl.$parsers.push(function(value){
           if (value) {
-            return value.toISOString();
```
Comment (petebacondarwin) [inline modules/directives/date/date.js:91]: It looks like you have completely removed the ISO date stuff altogether.

Comment (ProLoser) [inline modules/directives/date/date.js:91]: line 105 in this commit

File: modules/directives/date/date.js (hunk 1 above)
Comment (ProLoser) [inline modules/directives/date/date.js]: good catch

**Comment by ProLoser** [comment]
Corrected the improper variable use, however I think toISOString() is still there

File: modules/directives/date/date.js (hunk 2 above)
Comment (petebacondarwin) [inline modules/directives/date/date.js:91]: :+1:

**Comment by petebacondarwin** [comment]
Just working on the unit tests now

**Comment by petebacondarwin** [comment]
@ProLoser I have refactored the ui-date-format tests to isolate them from ui-date and added tests for uiConfig use.

**Comment by ProLoser** [comment]
Sweet thanks! I'm just going to merge them in as-is via github.

**Pull request merged**